other/dpll.py

```diff
@@ -0,0 +1,402 @@
+            if j in (True, None):
+                return j
+        for j in self.literals.values():
+            result = result or j
+        return result
+
+
+class Formula:
+    """
+    A formula represented in CNF.
+    A formula is a set of clauses.
+    For example,
+        {{A1, A2, A3'}, {A5', A2', A1}} is ((A1 v A2 v A3') and (A5' v A2' v A1))
+
+    Create two clauses and a formula with them
+    >>> clause1 = Clause(["A1", "A2'", "A3"])
+    >>> clause2 = Clause(["A5'", "A2'", "A1"])
+
+    >>> formula = Formula([clause1, clause2])
+    >>> print(formula)
+    {{ A1 , A2' , A3 } , { A5' , A2' , A1 }}
+    """
+
+    def __init__(self, clauses: List[Clause]) -> None:
+        """
+        Represent the number of clauses and the clauses themselves.
+        """
+        self.clauses = [c for c in clauses]
+        self.no_of_clauses = len(self.clauses)
+
+    def __str__(self) -> str:
+        """
+        To print a formula as in CNF.
+        Variable formula holds the string representation.
+        """
+        formula = "{"
+        clause_repr = " , "
+        clauses_as_strings = [str(clause) for clause in self.clauses]
+
+        clause_repr = clause_repr.join(clauses_as_strings)
+
+        formula += clause_repr
+        formula += "}"
+
+        return formula
+
+
+def generate_clause() -> Clause:
+    """
+    Randomly generate a clause.
+    All literals have the name Ax, where x is an integer from 1 to 5.
+    """
+    literals = []
+    no_of_literals = random.randint(1, 5)
+    base_var = "A"
+    i = 0
+    while i < no_of_literals:
+        var_no = random.randint(1, 5)
+        var_name = base_var + str(var_no)
+        var_complement = random.randint(0, 1)
+        if var_complement == 1:
+            var_name += "'"
+        if var_name in literals:
+            i -= 1
+        else:
+            literals.append(var_name)
+        i += 1
+    return Clause(literals)
+
+
+def generate_formula() -> Formula:
+    """
+    Randomly generate a formula.
+    """
+    clauses = []
+    no_of_clauses = random.randint(1, 10)
+    i = 0
+    while i < no_of_clauses:
+        clause = generate_clause()
+        if clause in clauses:
+            i -= 1
+        else:
+            clauses.append(clause)
+        i += 1
+    return Formula(set(clauses))
+
+
+def generate_parameters(formula: Formula) -> (List[Clause], List[str]):
+    """
+    Return the clauses and symbols from a formula.
+    A symbol is the uncomplemented form of a literal.
+    For example,
+        Symbol of A3 is A3.
+        Symbol of A5' is A5.
+
+    >>> clause1 = Clause(["A1", "A2'", "A3"])
+    >>> clause2 = Clause(["A5'", "A2'", "A1"])
+
+    >>> formula = Formula([clause1, clause2])
+    >>> clauses, symbols = generate_parameters(formula)
+    >>> clauses_list = [str(i) for i in clauses]
+    >>> print(clauses_list)
+    ["{ A1 , A2' , A3 }", "{ A5' , A2' , A1 }"]
+    >>> print(symbols)
+    ['A1', 'A2', 'A3', 'A5']
```

Comment:
```suggestion
    >>> clauses
    x, y, z
    >>> symbols
    ['A1', 'A2', 'A3', 'A5']
```
Do not post-process the outputs.  Do not print the outputs.  Just evaluate them the way that you would do in the Python REPL.